larrecodnn/ImagePatternAlgs/Tensorflow/Modules/RawWaveformDump_module.cc

```diff
@@ -0,0 +1,500 @@
+  std::string	fDigitModuleLabel; ///< module that made digits
+
+  std::string fSelectGenLabel;
+  std::string fSelectProcID;
+  int fSelectPDGCode;
+  std::string fPlaneToDump;
+  double fMinParticleEnergyGeV;
+  double fMinEnergyDepositedMeV;
+  int fMinNumberOfElectrons;
+  int fMaxNumberOfElectrons;
+  bool fSaveSignal;
+  art::ServiceHandle<geo::Geometry> fgeom;
+  art::ServiceHandle<cheat::ParticleInventoryService> PIS;
+  //art::ServiceHandle<SimChannelExtractService> m_pscx;
+  detinfo::DetectorClocks const * fClks;
+
+  std::default_random_engine rndm_engine;
+
+  c2numpy_writer npywriter;
+};
+
+//-----------------------------------------------------------------------
+struct genFinder{
+private:
+  typedef std::pair<int, std::string> track_id_to_string;
+  std::vector<track_id_to_string> track_id_map;
+  std::set<std::string> generator_names;
+  bool isSorted=false;
+
+public:
+  void sort_now(){
+    std::sort(this->track_id_map.begin(), this->track_id_map.end(), [](const auto &a, const auto &b){return (a.first < b.first) ; } );
+    isSorted=true;
+  }
+  void add(const int& track_id, const std::string& gname){
+    this->track_id_map.push_back(std::make_pair(track_id, gname));
+    generator_names.emplace(gname);
+    isSorted=false;
+  }
+  bool has_gen(std::string gname){
+    return static_cast<bool>(generator_names.count(gname));
+  };
+  std::string get_gen(int tid){
+    if( !isSorted ){
+      this->sort_now();
+    }
+    return std::lower_bound(track_id_map.begin(), track_id_map.end(), tid,[](const auto &a, const auto &b){return (a.first < b) ; } )->second;
+  };
+
+};
+genFinder* gf = new genFinder();
+
+//-----------------------------------------------------------------------
+nnet::RawWaveformDump::RawWaveformDump(fhicl::ParameterSet const& p)
+  : EDAnalyzer{p},
+  fClks(lar::providerFrom<detinfo::DetectorClocksService>())
+  {
+    this->reconfigure(p);
+  }
+
+//-----------------------------------------------------------------------
+void nnet::RawWaveformDump::reconfigure(fhicl::ParameterSet const & p)
+{
+  fDumpWaveformsFileName = p.get<std::string>("DumpWaveformsFileName","dumpwaveforms");
+  if (std::getenv("PROCESS")){
+    fDumpWaveformsFileName += string(std::getenv("PROCESS"))+"-";
+  }
+  fSimulationProducerLabel = p.get<std::string>("SimulationProducerLabel", "largeant");
+  fDigitModuleLabel = p.get<std::string>("DigitModuleLabel", "daq");
+
+  fSelectGenLabel = p.get<std::string>("SelectGenLabel","ANY");
+  fSelectProcID   = p.get<std::string>("SelectProcID",  "ANY");
+  fSelectPDGCode  = p.get<int>("SelectPDGCode", 0);
+
+  fPlaneToDump = p.get<std::string>("PlaneToDump", "U");
+  fMinParticleEnergyGeV = p.get<double>("MinParticleEnergyGeV",0.);
+  fMinEnergyDepositedMeV = p.get<double>("MinEnergyDepositedMeV",0.);
+  fMinNumberOfElectrons = p.get<int>("MinNumberOfElectrons",1000);
+  fMaxNumberOfElectrons = p.get<int>("MaxNumberOfElectrons",100000);
+
+  fSaveSignal = p.get<bool>("SaveSignal", true);
```

Comment:
These variables could be constructed in the initialisation list

`nnet::RawWaveformDump::RawWaveformDump(fhicl::ParameterSet const& p)
   : EDAnalyzer{p},
     fSimulationProducerLabel(p.get<std::string>("SimulationProducerLabel", "largeant")),
.....
     `

Reply:
This is now addressed.